File: TimeCardClientAndroid/app/src/main/java/jp/uhimania/timecardclientandroid/data/source/LocalTimeRecord.kt
package jp.uhimania.timecardclientandroid.data.source

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.Date

@Entity(
    tableName = "break_times",
    foreignKeys = [
        ForeignKey(
            entity = LocalTimeRecord::class,
            parentColumns = ["id"],
            childColumns = ["timeRecordId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class LocalBreakTime(
    @PrimaryKey val id: String = "",
    val start: Date?,
    val end: Date?,
    val timeRecordId: String
)

@Entity(tableName = "time_records", indices = [Index(value = ["year", "month"])])
data class LocalTimeRecord(
    @PrimaryKey val id: String = "",
    val year: Int,
    val month: Int,
    val checkIn: Date?,
    val checkOut: Date?
)

File: TimeCardClientIOS/TimeCardClientIOS/Data/Source/LocalTimeRecord.swift
//
//  LocalTimeRecord.swift
//  TimeCardClientIOS
//
//  Created by uhimania on 2026/03/11.
//

import Foundation
import SwiftData

@Model
class LocalTimeRecord {
    @Model
    class LocalBreakTime {
        var id: UUID
        var start: Date?
        var end: Date?
        var parent: LocalTimeRecord?
        
        init(id: UUID, start: Date? = nil, end: Date? = nil, parent: LocalTimeRecord? = nil) {
            self.id = id
            self.start = start
            self.end = end
            self.parent = parent
        }
    }
    
    #Index<LocalTimeRecord>([\.id], [\.year, \.month])
    
    var id: UUID
    var year: Int
    var month: Int
    var checkIn: Date?
    var checkOut: Date?
    
    @Relationship(deleteRule: .cascade, inverse: \LocalBreakTime.parent)
    var breakTimes: [LocalBreakTime]
    
    init(id: UUID, year: Int, month: Int, checkIn: Date? = nil, checkOut: Date? = nil, breakTimes: [LocalBreakTime]) {
        self.id = id
        self.year = year
        self.month = month
        self.checkIn = checkIn
        self.checkOut = checkOut
        self.breakTimes = breakTimes
    }
}
